packages/trees/frontend/sirius-components-trees/src/treeitems/filterTreeItem.ts

```diff
@@ -12,19 +12,48 @@
  *******************************************************************************/
 import { GQLTreeItem } from '../views/ExplorerView.types';
 
-export const isFilterCandidate = (treeItem: GQLTreeItem, highlightRegExp: RegExp): boolean => {
+export const splitText = (label: string, userInput: string): string[] => {
+  if (!userInput) {
+    return [label];
+  }
+
+  // Split the label in a case insensitive manner
+  const caseInsensitiveSplitLabel: string[] = label
+    .toLowerCase()
+    .split(userInput.toLocaleLowerCase())
+    .flatMap((value, index) => {
+      if (index === 0) {
+        return [value];
+      }
+      return [userInput.toLocaleLowerCase(), value];
+    });
+
+  // Create the real result
+  const splitLabel: string[] = [];
+  let index = 0;
+  for (const caseInsensitiveSegment of caseInsensitiveSplitLabel) {
+    if (caseInsensitiveSegment.length > 0) {
+      const caseSensitiveSegment = label.substring(index, index + caseInsensitiveSegment.length);
+      splitLabel.push(caseSensitiveSegment);
+      index = index + caseInsensitiveSegment.length;
+    }
+  }
+  return splitLabel;
+};
+
+export const isFilterCandidate = (treeItem: GQLTreeItem, textToHighlight: string | null): boolean => {
   let filter: boolean = false;
-  const splitLabelWithTextToHighlight: string[] = treeItem.label.split(highlightRegExp);
+  const splitLabelWithTextToHighlight: string[] = splitText(treeItem.label, textToHighlight);
   if (splitLabelWithTextToHighlight.length > 1) {
     filter = false;
   } else if (!treeItem.hasChildren && splitLabelWithTextToHighlight.length === 1) {
     filter = true;
   } else if (
     treeItem.hasChildren &&
     treeItem.expanded &&
-    treeItem.children.map((child) => child.label.split(highlightRegExp).length).every((v) => v === 1)
+    treeItem.children.map((child) => child.label.split(textToHighlight).length).every((v) => v === 1)
```

Comment:
May be we should also use `splitText` here, because `split` will be sensitive-case. What do you think about?

Reply:
The issue in reusing spliText here is that it would also reinsert the separator (the part with flatMap), we should probably just use toLowerCase. Could you update the PR for that?

Reply:
Ok I'll do that